model/llm/llm_test.go

```diff
@@ -340,3 +340,279 @@ func TestLLMCodeRepairSourceFilePrompt(t *testing.T) {
+			package com.eval;
+
+			class Foobar {
+				static int foobar(int i) {}
+			}
+			` + "```" + `
+		`),
+	})
+}
+
+func TestModelTranspile(t *testing.T) {
+	type testCase struct {
+		Name string
+
+		SetupMock func(t *testing.T, mockedProvider *providertesting.MockQuery)
+
+		Language       language.Language
+		TargetLanguage language.Language
+
+		RepositoryPath string
+		SourceFilePath string
+		StubFilePath   string
+
+		ExpectedAssessment      metrics.Assessments
+		ExpectedStubFileContent string
+	}
+
+	validate := func(t *testing.T, tc *testCase) {
+		logOutput, logger := log.Buffer()
+		defer func() {
+			if t.Failed() {
+				t.Log(logOutput.String())
+			}
+		}()
+
+		temporaryPath := t.TempDir()
+		repositoryPath := filepath.Join(temporaryPath, filepath.Base(tc.RepositoryPath))
+		require.NoError(t, osutil.CopyTree(tc.RepositoryPath, repositoryPath))
+
+		modelID := "some-model"
+		mock := providertesting.NewMockQuery(t)
+		tc.SetupMock(t, mock)
+		llm := NewModel(mock, modelID)
+
+		ctx := model.Context{
+			Language: tc.Language,
+
+			RepositoryPath: repositoryPath,
+			FilePath:       tc.SourceFilePath,
+
+			Arguments: &evaluatetask.TaskArgumentsTranspile{
+				TargetLanguage: tc.TargetLanguage,
+				StubFilePath:   tc.StubFilePath,
+			},
+
+			Logger: logger,
+		}
+
+		actualAssessment, actualError := llm.Transpile(ctx)
+		assert.NoError(t, actualError)
+		metricstesting.AssertAssessmentsEqual(t, tc.ExpectedAssessment, actualAssessment)
+
+		actualStubFileContent, err := os.ReadFile(filepath.Join(repositoryPath, tc.StubFilePath))
+		assert.NoError(t, err)
+
+		assert.Equal(t, strings.TrimSpace(bytesutil.StringTrimIndentations(tc.ExpectedStubFileContent)), string(actualStubFileContent))
+	}
+
+	t.Run("Transpile Go into Java", func(t *testing.T) {
+		stubFileContent := `
+			package com.eval;
+
+			class BinarySearch {
+				static int binarySearch(int[] a, int x) {
+					int index = -1;
+
+					int min = 0;
+					int max = a.length - 1;
+
+					while (index == -1 && min <= max) {
+						int m = (min + max) / 2;
+
+						if (x == a[m]) {
+							index = m;
+						} else if (x < a[m]) {
+							max = m - 1;
+						} else {
+							min = m + 1;
+						}
+					}
+
+					return index;
+				}
+			}
+		`
+		validate(t, &testCase{
+			Name: "Binary search",
+
+			SetupMock: func(t *testing.T, mockedProvider *providertesting.MockQuery) {
+				mockedProvider.On("Query", mock.Anything, "some-model", mock.Anything).Return(bytesutil.StringTrimIndentations(`
+					`+"```"+`
+					package com.eval;
+
+					class BinarySearch {
+						static int binarySearch(int[] a, int x) {
+							int index = -1;
+
+							int min = 0;
+							int max = a.length - 1;
+
+							while (index == -1 && min <= max) {
+								int m = (min + max) / 2;
+
+								if (x == a[m]) {
+									index = m;
+								} else if (x < a[m]) {
+									max = m - 1;
+								} else {
+									min = m + 1;
+								}
+							}
+
+							return index;
+						}
```

Comment:
can use `stubFileContent`

Reply:
Tried to, but using it like:

```go
mockedProvider.On("Query", mock.Anything, "some-model", mock.Anything).Return(bytesutil.StringTrimIndentations(`
					`+"```"+`
					`+stubFileContent+`
					`+"```"+`
				`), nil)
```

errors in the bytesutil function with missing indentation errors.